File: Array/SubarrayWithGivenSum/efficientApproach.js
// function isSubSum(arr, sum) {
//   let s = 0,
//     curr = 0;
//   for (let e = 0; e < arr.length; e++) {
//     curr += arr[e];
//     while (sum < curr) {
//       curr -= arr[s];
//       s++;
//     }
//     if (curr === sum) return true;
//   }
//   return false;
// }
// console.log(isSubSum([1, 4, 20, 3, 10, 5], 33));

File: Array/PrefixSum/efficientApproach.cpp
#include <iostream>
using namespace std;

int prefixSum(int arr[1000], int n, int startingIndex, int endingIndex)
{
    int res = 0;
    for (int i = startingIndex; i < endingIndex + 1; i++)
    {
        res += arr[i];
    }
    return res;
}
int main()
{
    int arr[1000], startingIndex, endingIndex, n, res;
    cout << "Enter the number of elements  in array: ";
    cin >> n;
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    cout << "Enter starting index then ending index-> ";
    cin >> startingIndex >> endingIndex;

    res = prefixSum(arr, n, startingIndex, endingIndex);
    cout << "Prefix sum is-> " << res << endl;
    return 0;
}